refinery/tests/postgres.rs

```diff
@@ -65,35 +67,44 @@ mod postgres {
     }
 
     fn clean_database() {
-        let mut client =
-            Client::connect("postgres://postgres@localhost:5432/template1", NoTls).unwrap();
+        let uri = db_uri();
+        let db_name = uri.split('/').last().unwrap();
+
+        let mut client = Client::connect(
+            &(uri.strip_suffix(db_name).unwrap().to_string() + "template1"),
+            NoTls,
+        )
+        .unwrap();
 
         client
             .execute(
-                "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname='postgres'",
-                &[],
+                "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname=$1",
+                &[&db_name],
             )
             .unwrap();
-        client.execute("DROP DATABASE POSTGRES", &[]).unwrap();
-        client.execute("CREATE DATABASE POSTGRES", &[]).unwrap();
+        client
+            .execute(&"DROP DATABASE IF EXISTS $1".replace("$1", db_name), &[])
+            .unwrap();
+        client
+            .execute(&"CREATE DATABASE $1".replace("$1", db_name), &[])
+            .unwrap();
     }
 
     fn run_test<T>(test: T)
     where
         T: FnOnce() + std::panic::UnwindSafe,
     {
-        let result = std::panic::catch_unwind(test);
-
```

Comment:
do we need to move the `catch_unwind`?

Reply:
Yeah, it could probably go, and it'd be worth renaming that method to something more like `setup_database`, rather than `clean_database`, to better reflect its new purpose.  In fact, I've also recently seen a way to clean a DB that doesn't require admin privs, so I might put that in there, too...

Reply:
thanks!